autoreject/autoreject.py

```diff
@@ -530,38 +490,193 @@ def _interpolate_bad_epochs(self, epochs, verbose='progressbar'):
+                                good_epochs_idx, axis=0)
+        del epochs_copy  # I can't wait for garbage collection.
+        return self
+
+    def transform(self, epochs):
+        """Fix and find the bad epochs.
+
+        .. note::
+           LocalAutoReject partially supports multiple channels.
+           While fitting, at this point requires selection of channel types,
+           the transform can handle multiple channel types, if `.threshes_`
+           parameter contains all necessary channels and `.consensus_perc`
+           and `n_interpolate` have meaningful channel type specific
+           settings. These are commonly obtained from
+           :func:`autoreject.LocalAutoRejectCV`.
+
+        Parameters
+        ----------
+        epochs : instance of mne.Epochs
+            The epochs object for which bad epochs must be found.
+        """
+        _check_data(epochs, picks=self.picks, verbose=self.verbose,
+                    ch_constraint='data_channels')
+        if not all(epochs.ch_names[pp] in self.threshes_ for pp in self.picks):
+            raise ValueError('You are passing channels which were not present '
+                             'at fit-time. Please fit it again, this time '
+                             'correctly.')
+        epochs_out = epochs.copy()
+        sub_picks = _check_sub_picks(picks=self.picks, info=epochs_out.info)
+        if sub_picks is not False:
+            bad_epochs_idx = list()
+            bad_channels_list = list()
+            old_picks = self.picks
+            for ii, (ch_type, this_picks) in enumerate(sub_picks):
+                self.picks = this_picks
+                out = self._annotate_epochs(self.threshes_, epochs)
+                bad_channels_list.append(out[2])
+                bad_epochs_idx_ = out[4]
+                bad_epochs_idx = np.union1d(bad_epochs_idx, bad_epochs_idx_)
+            good_epochs_idx = np.setdiff1d(np.arange(len(epochs)),
+                                           bad_epochs_idx)
+            if len(good_epochs_idx) == 0:
+                raise ValueError('All epochs are bad. Sorry.')
+
+            for ii, (ch_type, this_picks) in enumerate(sub_picks):
+                self.picks = this_picks
+                self._interpolate_bad_epochs(
+                    epochs_out, bad_channels=bad_channels_list[ii],
+                    verbose=self.verbose)
+            self.picks = old_picks
+
+        else:
+            (_, _, bad_channels, _,
+             bad_epochs_idx, good_epochs_idx) = self._annotate_epochs(
+                 threshes=self.threshes_, epochs=epochs)
+            if len(good_epochs_idx) == 0:
+                raise ValueError('All epochs are bad. Sorry.')
+
+            self._interpolate_bad_epochs(
+                epochs_out, bad_channels=bad_channels,
+                verbose=self.verbose)
+
+        epochs_out.drop(bad_epochs_idx, reason='AUTOREJECT')
+        return epochs_out
+
+    def _interpolate_bad_epochs(
```

Comment:
actually it's great that you refactored this. In the near future, this can be handled by MNE